contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {IMakerRouter} from "./IMakerRouter.sol";
+import {IDSSPSM} from "./IDSSPSM.sol";
+import {IPegStabilityModule} from "../../peg/IPegStabilityModule.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {CoreRef} from "../../refs/CoreRef.sol";
+import {TribeRoles} from "../../core/TribeRoles.sol";
+import {Constants} from "../../Constants.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+/// @notice This contracts allows for swaps between FEI-DAI and FEI-USDC
+/// by using the FEI FEI-DAI PSM and the Maker DAI-USDC PSM
+/// @author k-xo
+contract MakerRouter is IMakerRouter, CoreRef {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the Maker DAI-USDC PSM that this router interacts with
+    /// @dev points to Makers DssPsm contract
+    IDSSPSM public immutable daiPSM;
+
+    /// @notice reference to the FEI FEI-DAI PSM that this router interacts with
+    IPegStabilityModule public immutable feiPSM;
+
+    /// @notice reference to the DAI contract used.
+    /// @dev Router can be redeployed if DAI address changes
+    IERC20 public immutable dai;
+
+    /// @notice reference to the FEI contract used.
+    /// @dev Router can be redeployed if FEI address changes
+    IERC20 public immutable fei;
+
+    /// @notice scaling factor for USDC
+    uint256 public constant USDC_SCALING_FACTOR = 1e12;
+
+    constructor(
+        address _core,
+        IDSSPSM _daiPSM,
+        IPegStabilityModule _feiPSM,
+        IERC20 _dai,
+        IERC20 _fei
+    ) CoreRef(_core) {
+        daiPSM = _daiPSM;
+        feiPSM = _feiPSM;
+
+        dai = _dai;
+        fei = _fei;
+
+        fei.approve(address(feiPSM), type(uint256).max);
```

Comment:
Would really avoid issuing an infinite approval where possible. This contract will only be called sporadically, so I would just approve the relevant amount of fei/dai when needed

Reply:
This contract is going to be limited to only allowing restricted roles within the system being able to call any state changing function. The FEI PSM and the Maker PSM are both not upgradeable so we should be fine.

Reply:
I agree that infinite approvals are something we don't want to frequently use as they are generally an anti-pattern.